### PR TITLE
Add dynamic string array

### PR DESCRIPTION
**Pull request by silvanadomnica**

### CODE REVIEW

File: silvanadomnica_Task2/DynamicStringArray.cpp [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            for (int j = 0; j < i; ++j) {
+                newArray[j] = dynamicArray[j];
+            }
+            for (int j = i+1; j < size; ++j) {
+                newArray[j-1] = dynamicArray[j];
+            }
+            delete[] dynamicArray;
+            dynamicArray = newArray;
+            --size;
+            return true;
```
Comment (maxlig79) [inline silvanadomnica_Task2/DynamicStringArray.cpp]: Please avoid multiple return.

File: silvanadomnica_Task2/DynamicStringArray.cpp [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+DynamicStringArray::~DynamicStringArray()
+{
+    delete[] dynamicArray;
+}
+DynamicStringArray& DynamicStringArray::operator=(const DynamicStringArray &obj)
+{
+    if(&obj != this)
+    {
+        size=obj.size;
+        dynamicArray=new string[size];
```
Comment (maxlig79) [inline silvanadomnica_Task2/DynamicStringArray.cpp]: What happen if the dynamicArray is already allocated?

File: silvanadomnica_Task2/DynamicStringArray.cpp [hunk 3]
```diff
@@ -0,0 +1,79 @@
+#include "DynamicStringArray.hpp"
+
+DynamicStringArray::DynamicStringArray()
+{
```
Comment (maxlig79) [inline silvanadomnica_Task2/DynamicStringArray.cpp:4]: As a suggestion use initializer list, in general you have better performance.

**Review by maxlig79 — CHANGES_REQUESTED** [review]

File: silvanadomnica_Task2/DynamicStringArray.cpp [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        size--;
+    }
+
+    return found;
+}
+string DynamicStringArray::getEntry(const int index) const
+{
+    if (index >= size)
+    {
+        return nullptr;
```
Comment (maxlig79) [inline silvanadomnica_Task2/DynamicStringArray.cpp]: nullptr is not a string.

**Review by maxlig79 — CHANGES_REQUESTED** [review]

**Review by maxlig79 — APPROVED** [review]

**Review by nbiro — APPROVED** [review]

**Pull request merged**